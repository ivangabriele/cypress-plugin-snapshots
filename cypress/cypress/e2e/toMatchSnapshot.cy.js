const { runSuites } = require('../fixtures/test-data');

describe('toMatchSnapshot', () => {
  it('html escaping', () => {
    cy.visit('/static/input.html')
      .then(() => {
        cy.get('#input-element').toMatchSnapshot();
      });
  });

  it('custom name', () => {
    cy.wrap('test').toMatchSnapshot({
      name: 'screenshot with custom name'
    });
  });

  runSuites('toMatchSnapshot', function (item) {
    cy.wrap(item).toMatchSnapshot();
  }, ['date']);
});
