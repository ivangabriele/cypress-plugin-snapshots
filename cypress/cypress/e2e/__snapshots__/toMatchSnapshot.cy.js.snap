exports[`toMatchSnapshot > toMatchSnapshot - json #0`] =
{
  "number": 1234,
  "object": {
    "foo": "bar",
    "nestedObject": {
      "bar": "foo"
    }
  },
  "string": "string\\d"
};

exports[`toMatchSnapshot > toMatchSnapshot - html #0`] = `
<div class="wrapper" data-test="test">
  <h1>Lorem \\d Ipsum</h1>
  <p>
    Lorem ipsum dolor sit amet, <a href="#">consectetur</a> adipiscing elit.
    Integer dignissim, odio sed interdum porta, urna ex ultricies tellus, non
    fringilla purus nulla nec augue. Vivamus eget iaculis lorem. Proin blandit
    dictum enim, vitae efficitur arcu euismod a. Sed sodales nisi volutpat dui
    convallis ullamcorper. Aenean eget semper metus. Curabitur quis imperdiet
    lorem. Ut vitae mauris augue. Pellentesque habitant morbi tristique senectus
    et netus et malesuada fames ac turpis egestas. Ut rhoncus augue et
    scelerisque malesuada. Pellentesque bibendum sit amet tellus sit amet
    pellentesque. Maecenas eu magna dapibus, tristique est nec, ornare libero.
    Sed consectetur consequat nisi.
  </p>
  <div data-test="test2">
    <h2>Curabitur finibus quam</h2>
    <p>
      Curabitur finibus quam quam, in viverra nulla dapibus sed. Praesent cursus
      nec erat ut tempor. Donec pharetra rutrum aliquam. Mauris eget gravida
      quam. Aenean sodales eros a <a href="#">felis lobortis</a> faucibus. Donec
      sed luctus purus. Nullam nisi ipsum, scelerisque at felis eu, aliquam
      condimentum ex. Quisque nec orci at justo faucibus maximus. Suspendisse
      lobortis id mauris in pellentesque. Praesent et dolor ultricies,
      vestibulum ipsum vel, auctor lacus. Nullam nec dui id arcu consectetur
      aliquam.
    </p>
  </div>
</div>
`;

exports[`toMatchSnapshot > toMatchSnapshot - html escaping #0`] = `
<span class="input input--number input--required" id="input-element">
  <input
    type="number"
    name="new_address.house_number"
    value=""
    pattern="\\d*"
    required=""
    data-date-format="mm/dd/yyyy"
  />
</span>
`;

exports[`toMatchSnapshot > toMatchSnapshot > json > true #0`] =
true;

exports[`toMatchSnapshot > toMatchSnapshot > json > number #0`] =
5;

exports[`toMatchSnapshot > toMatchSnapshot > json > array #0`] =
[
  1,
  "string",
  false,
  null,
  {
    "a": 1,
    "b": 2
  },
  null,
  11
];

exports[`toMatchSnapshot > toMatchSnapshot > json > object #0`] =
{
  "array": [
    "string",
    5,
    true,
    null,
    {
      "a": 1,
      "b": 2
    },
    null
  ],
  "false": false,
  "nested": {
    "array": [
      2,
      "string",
      true,
      null,
      null,
      {
        "a": 1,
        "b": 2
      }
    ],
    "number": 1,
    "true": true
  },
  "null": null,
  "string": "string"
};

exports[`toMatchSnapshot > toMatchSnapshot > html > jquery-get #0`] = `
<h4>H1</h4>
<h4>H2</h4>
<h4>H3</h4>
`;

exports[`toMatchSnapshot > toMatchSnapshot > html > jquery-create #0`] = `
<div>Created DIV</div>
`;

exports[`toMatchSnapshot > toMatchSnapshot > html > cy-get-element #0`] = `
<h4>H1</h4>
<h4>H2</h4>
<h4>H3</h4>
`;

exports[`toMatchSnapshot > toMatchSnapshot > html > DOM-exists #0`] = `
<div id="test-div">Test Div</div>
`;

exports[`toMatchSnapshot > toMatchSnapshot > html > DOM-created #0`] = `
<div>DOM-created</div>
`;

exports[`toMatchSnapshot > toMatchSnapshot > html > HTMLCollection #0`] = `
<h4>H1</h4>
<h4>H2</h4>
<h4>H3</h4>
`;

exports[`toMatchSnapshot > toMatchSnapshot > html > NodeList #0`] = `
<h4>H1</h4>
<h4>H2</h4>
<h4>H3</h4>
`;

exports[`toMatchSnapshot > toMatchSnapshot > html > svg #0`] = `
<svg height="20" width="40"><path d="M10 0 L0 20 L20 20 Z"></path></svg>
`;

exports[`toMatchSnapshot > toMatchSnapshot > json > false #0`] =
false;

exports[`toMatchSnapshot > toMatchSnapshot > json > null #0`] =
null;

exports[`toMatchSnapshot > toMatchSnapshot > json > string #0`] = `
kylie
`;

exports[`toMatchSnapshot > toMatchSnapshot > json > date #0`] = `
2019-02-11T05:00:00.000Z
`;

exports[`toMatchSnapshot > custom name #0`] = `
test
`;
